### ISSUE TITLE
Stack space for flow elements

### ISSUE THREAD

**Issue by thewebartisan7** [twbs/bootstrap#32034]
This feature is best explained here:
https://every-layout.dev/layouts/stack/

And an example can be found here:
https://jsfiddle.net/c2fbo0u3/1/

In the example I have just added two different space size, 1.5rem and large 3rem, but could be also added more.

As explained in every-layout.dev this is useful when we need to add space between elements, and in addition I added space X and Y, something like already bootstrap m-* utilities offer.

So instead of writing:

```html
<div>
<div></div>
<div class="mt-3"></div>
<div class="mt-3"></div>
</div>
```

Can be:

```html
<div class="stack-y">
<div></div>
<div></div>
<div></div>
</div>
```

Or instead of:

```html
<div class="d-flex">
<div></div>
<div class="ml-3"></div>
<div class="ml-3"></div>
</div>
```

Can be:

```html
<div class="d-flex stack-x">
<div></div>
<div></div>
<div></div>
</div>
```

Here https://jsfiddle.net/c2fbo0u3/1/ there are all examples, also creating a reverse space for space-x, and nested.

Nested could be also created without using space-nested but just adding for each nested element the same space-x/y classes. Example:

```html
<div class="stack-x stack-nested d-flex">
  <div>Stack nested X One</div>
  <div>Stack nested X Two</div>
  <div class="d-flex">
    <div>Stack nested X One</div>
    <div>Stack nested X Two</div>
    <div>Stack nested X Three</div>
  </div>
</div>
```

Can be:

```html
<div class="stack-x d-flex">
  <div>Stack nested X One</div>
  <div>Stack nested X Two</div>
  <div class="stack-x d-flex">
    <div>Stack nested X One</div>
    <div>Stack nested X Two</div>
    <div>Stack nested X Three</div>
  </div>
</div>
```

**Comment by mdo** [comment]
This is very familiar to [Pylon's hstack/vstack approach](https://almonk.github.io/pylon/). Very intriguing to me also considering #31995 and not wanting to have a single component-based layout option. I'd go further than here and suggest that it be `display: flex` by default and offer inline-spacers via CSS customer properties. Example: https://codepen.io/emdeoh/pen/Bazxbrq.

**Comment by thewebartisan7** [comment]
Yes I agree to be already display flex, that is how I setup mine custom at the moment. 
I didn't mention but using already display flex, for hstack (btw I like so much this name .vstack / .hstack that you use), then can be possible to create also reversed order. At the moment I have a .stack-x-reversed (or better .hstack-reversed)

```
.stack-x-reverse {
    display: flex;
    flex-direction: row;
    justify-content: flex-start;
    > * { margin-left: 0; margin-right: 0; }
    > * + * { margin-right: 1.5rem; }
    &.stack-lg > * + * { margin-right: 3rem; }
    &.stack-nested {
        * { margin-left: 0; margin-right: 0; }
        * + * { margin-right: 1.5rem; }
        &.stack-lg * + * { margin-right: 3rem; }
    }
}
```

I like also idea of css custom properties inline, so we don't need another set of margins. But anyway, most of time I use only 2spaces, small and large.

**Comment by mdo** [comment]
Our new `.d-grid` and `.gap` utilities make this a lot easier for vstack variations. hstack would still theoretically require more code, but a `.d-grid-row` helper class could provide that. https://codepen.io/emdeoh/pen/VwjEgog

**Comment by mdo** [comment]
Check out #32320.

**Comment by ffoodd** [comment]
For the record, in progress in #33986 :)

**Comment by XhmikosR** [comment]
Fixed by #33986
